core/transaction_pool.go

```diff
@@ -71,7 +71,12 @@ type TxPool struct {
 	queue   map[common.Address]map[common.Hash]*types.Transaction
 }
 
-func NewTxPool(eventMux *event.TypeMux, currentStateFn stateFn, gasLimitFn func() *big.Int) *TxPool {
+func NewTxPool(eventMux *event.TypeMux, currentStateFn stateFn, gasLimitFn func() *big.Int) (*TxPool, error) {
+	currentState, err := currentStateFn()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Can this actually fail though - ```currentStateFn``` is currently always ```eth.blockchain.State``` which should always succeed, right? It can only fail if ```blockchain``` has a inconsistent state where it's current block's state root is not present in it's db.